src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/BundledAppWithSubDirs.cs

```diff
@@ -59,6 +59,8 @@ public void Bundled_Framework_dependent_App_Run_Succeeds(BundleOptions options)
         public void Bundled_Self_Contained_App_Run_Succeeds(BundleOptions options)
         {
             var fixture = sharedTestState.TestSelfContainedFixture.Copy();
+            UseSingleFileSelfContainedHost(fixture);
+            options |= BundleOptions.EnableCompression;
             var singleFile = BundleHelper.BundleApp(fixture, options);
```

Comment:
Given that we plan to have this on by default - I would probably change the default test behavior as well. Make the `BundleHelper` turn on compression by default and have a way for the test to selectively disable it.

Reply:
I agree to a point - basically if I add a new test - I will definitely forget to set this option - and thus my test will not cover the most common scenario. Given the history of bad and few tests in this area I'd like to make it so that we have at least reasonable coverage for the most common cases.

Reply:
I have explored making `UseSingleFileSelfContainedHost` to enable compression automatically, but it was not very natural since the “fixture” stuff operates at proj file level and does not know much about the bundle.

Perhaps I should revisit that direction. Maybe set a flag on the fixture, that BundleHelper would read.

Reply:
Nope going further that direction only make things worse. If `UseSingleFileSelfContainedHost` sets "can do compression" flag, then there are two ways to specify whether compression is desired - because fixture says so and via flags passed to bundler. The flag on the fixture must take precedence and that is a bit unintuitive and could be error prone if fixture is reused. Also for the unusual/error cases I would need another argument to override the override. It gets messy.

Here what seems to work better.
- Add a helper `BundleSelfContainedApp` which both picks the right host and then adds compression flags and passes that to Bundler.
- Make all tests that could use `BundleSelfContainedApp` use it - that is nearly all tests that do singlefile stuff.
- A a few tests that need to tests unusual/error combinations go the long route - they will have to use `UseSingleFileSelfContainedHost` and explicitly and have spell out what they need in terms of compression to `BundleApp`. There are only a few tests like that.

I would expect the obvious difference in verbosity will direct future testcase writers towards the "default" path.